src/PolykeyAgent.ts

```diff
@@ -562,6 +598,7 @@ class PolykeyAgent {
       await this.keyManager?.stop();
       await this.schema?.stop();
       await this.status?.stop({});
+      this.events.removeAllListeners(keysUtils.eventRootKeyPairChange);
```

Comment:
Is this in the right order? Order of shutdown should be opposite of order of startup.

So if A, then B, then C, then shutdown is C, then B, then A.

Reply:
Yeah the handlers are registered right at the top

Reply:
I have a feeling that event handlers should be done after all the domains are constructed. So they should be the first to be deregistered.

Reply:
And therefore you can refer to the objects, not using `this`, but just by referring to the objects in the function closure. Like instead of `this.fwdProxy.setTLSConfig`, you just do `fwdProxy.setTLSConfig`.

Reply:
They're being registered in `start`, so all of the domains are already constructed. Since `start` doesn't have access to the objects the only way to do it without the `this` would be to move the registering into `createPolykeyAgent`, in which case deregistering would go into `destroy` instead of `stop`.

Reply:
Ah yes, I forgot about that. Ok this is fine then.